src/Components/Classes/ClassForm/index.jsx

```diff
@@ -0,0 +1,279 @@
+      });
+    } catch (error) {
+      setResponseModal({ error: true, msg: error });
+      setShowModal(true);
+      throw new Error(error);
+    }
+  };
+  useEffect(() => {
+    getTrainersActivities();
+    !isCreateRoute && getById(id);
+  }, []);
+  useEffect(() => {
+    if (showModal && !responseModal.error) {
+      const timer = setTimeout(() => {
+        history.goBack();
+      }, 2000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [showModal, history]);
+
+  const createClass = async (newClass) => {
+    newClass.trainer = [newClass.trainer];
+    const bodyClasses = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newClass)
+    };
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/class/`, bodyClasses);
+      const data = await response.json();
+      if (data.length !== 0 && !data.error) {
+        setResponseModal({ error: false, msg: 'Class created sucessfully' });
+        setShowModal(true);
+      } else {
+        setResponseModal({ error: true, msg: data.message });
+        setShowModal(true);
+      }
+    } catch (error) {
+      setResponseModal({ error: true, msg: error });
+      setShowModal(true);
+      throw new Error(error);
+    }
+  };
+  const updateClass = async () => {
+    try {
+      const editedClass = {
+        ...classes,
+        trainer: [classes.trainer]
+      };
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/class/${id}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(editedClass)
+      });
+      console.log(response);
+      if (!response.ok) {
+        setResponseModal({ error: true, msg: response.data.message });
+        setShowModal(true);
+      }
+      setResponseModal({ error: false, msg: 'Class updated sucessfully' });
+      setShowModal(true);
+
+      return response.json();
+    } catch (error) {
+      setResponseModal({ error: true, msg: error });
+      setShowModal(true);
+      throw new Error(error.message);
+    }
+  };
+  const onClickEditClass = (e) => {
+    e.preventDefault();
+    if (
+      Object.values(classes).every((prop) => {
+        if (prop === '') {
+          return false;
+        }
+        return true;
+      })
+    ) {
+      updateClass();
+      setError(false);
+    } else {
+      setError(true);
+    }
+  };
+  const onClickCreateClass = (e) => {
+    e.preventDefault();
+    if (
+      Object.values(classes).every((prop) => {
+        if (prop === '') {
+          return false;
+        }
+        return true;
+      })
+    ) {
+      createClass(classes);
+      setError(false);
+    } else {
+      setError(true);
+    }
+  };
+  const cancelForm = (e) => {
+    e.preventDefault();
+    history.goBack();
+  };
+  const onChangeInput = async (e) => {
+    await setClasses({
```

Comment:
Hi, here there's no need for async/await in this onChangeInput function, because the state update already happens asynchronously, so its not necessary to set an await. This was explained to me in my previous PR.